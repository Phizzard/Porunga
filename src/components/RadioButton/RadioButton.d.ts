import React from 'react'
import { Palette } from '../../types'
/**
 * RadioButton properties
 */
export interface RadioButtonProps {
  /** aria-label will be added to the element and used by screen readers */
  ariaDescribedBy?: string
  /** A backgroundColor identifier */
  backgroundColor?: Palette
  /** A checked identifier */
  checked?: boolean
  /** Content inside the RadioButton element */
  children?: React.ReactNode
  /** Class attribute */
  className?: string
  /** The color of text using an identifier */
  color?: Palette
  /** A disabled identifier */
  disabled?: boolean
  /** A label identifier */
  label?: string
  /** A name identifier */
  name: string
  /** A value identifier */
  value: string
}
/**
 * RadioButton
 */
export declare const RadioButton: ({
  ariaDescribedBy,
  backgroundColor,
  checked,
  children,
  className,
  color,
  disabled,
  label,
  name,
  value,
}: RadioButtonProps) => JSX.Element
export default RadioButton
