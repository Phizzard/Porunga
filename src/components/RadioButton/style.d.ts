/// <reference types="react" />
/// <reference types="@emotion/core" />
interface RadioButtonProps {
  color?: string
  backgroundColor?: string
}
export declare const Container: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').LabelHTMLAttributes<HTMLLabelElement>,
    HTMLLabelElement
  >,
  RadioButtonProps,
  object
>
export declare const Input: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  >,
  RadioButtonProps,
  object
>
export {}
