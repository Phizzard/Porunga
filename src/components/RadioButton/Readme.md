Default RadioButton

```jsx
<RadioButton name="radio" value="one" label="One" />
<RadioButton name="radio" value="two" label="Two" />
<RadioButton name="radio" value="three" label="Three" />
```
