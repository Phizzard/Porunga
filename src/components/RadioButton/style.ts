import styled from '@emotion/styled'

interface RadioButtonProps {
  color?: string
  backgroundColor?: string
}

export const Container = styled.label<RadioButtonProps>`
  color: ${p => p.color};
  display: block;
  position: relative;
  padding-left: 2rem;
  margin-bottom: 0.5rem;
  cursor: pointer;
  font-size: 22px;
  user-select: none;

  /* Create a custom radio button */
  .checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border-radius: 50%;
  }

  /* Create the indicator (the dot/circle - hidden when not checked) */
  .checkmark:after {
    content: '';
    position: absolute;
    display: none;
  }

  /* Style the indicator (dot/circle) */
  .checkmark:after {
    top: 9px;
    left: 9px;
    width: 8px;
    height: 8px;
    border-radius: 50%;
    background: white;
  }
`

export const Input = styled.input<RadioButtonProps>`
  /* Hide the browser's default radio button */
  position: absolute;
  opacity: 0;
  cursor: pointer;

  /* On mouse-over, add a grey background color */
  :hover ~ .checkmark {
    background-color: #ccc;
  }

  /* When the radio button is checked, add a blue background */
  :checked ~ .checkmark {
    background-color: ${p => p.backgroundColor};
  }

  /* Show the indicator (dot/circle) when checked */
  :checked ~ .checkmark:after {
    display: block;
  }
`
