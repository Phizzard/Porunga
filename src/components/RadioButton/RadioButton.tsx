import React from 'react'
import { useTheme } from 'emotion-theming'

import { Palette, ThemeConfig } from '../../types'
import { get } from '../../utils/get'
import { Container, Input } from './style'

/**
 * RadioButton properties
 */
export interface RadioButtonProps {
  /** aria-label will be added to the element and used by screen readers */
  ariaDescribedBy?: string
  /** A backgroundColor identifier */
  backgroundColor?: Palette
  /** A checked identifier */
  checked?: boolean
  /** Content inside the RadioButton element */
  children?: React.ReactNode
  /** Class attribute */
  className?: string
  /** The color of text using an identifier */
  color?: Palette
  /** A disabled identifier */
  disabled?: boolean
  /** A label identifier */
  label?: string
  /** A name identifier */
  name: string
  /** A value identifier */
  value: string
}

/**
 * RadioButton
 */
export const RadioButton = ({
  ariaDescribedBy,
  backgroundColor = 'brand',
  checked,
  children,
  className,
  color = 'dark-1',
  disabled,
  label,
  name,
  value,
}: RadioButtonProps) => {
  const theme: ThemeConfig = useTheme()

  const radioButtonProps = {
    'aria-describedby': ariaDescribedBy,
    className,
    backgroundColor: get(theme, 'root', 'colors', backgroundColor),
    colorProp: get(theme, 'root', 'colors', color),
  }

  return (
    <Container>
      <Input
        checked={checked}
        disabled={disabled}
        name={name}
        type="radio"
        value={value}
        {...radioButtonProps}
      />
      <span className="checkmark"></span>
      {label}
      {children}
    </Container>
  )
}

export default RadioButton
