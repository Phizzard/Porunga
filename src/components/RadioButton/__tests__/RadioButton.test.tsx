import React from 'react'
import renderer from 'react-test-renderer'
import RadioButton from '../RadioButton'

it('renders correctly', () => {
  const component = renderer.create(
    <RadioButton name="radio" value="one" label="One"></RadioButton>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
