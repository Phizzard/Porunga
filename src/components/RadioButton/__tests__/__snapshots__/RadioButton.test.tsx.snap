// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="css-18vfld2"
>
  <input
    className="css-1glukks"
    name="radio"
    type="radio"
    value="one"
  />
  <span
    className="checkmark"
  />
  One
</label>
`;
